### PR TITLE
FoundationEssentials: implement file operations

### PR DESCRIPTION
**Pull request by compnerd**
This adds an initial implementation for file operations on Windows. With this, `FileManager` should at least build on Windows, which unblocks future work as well as brings us closer to enabling `FoundationEssentials` on Windows.

### CODE REVIEW

**Review by parkera — APPROVED** [review]
I'll leave the detailed review to @jmschonfeld but this looks good to me. Thanks for taking on this porting work @compnerd!

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (jmschonfeld) [inline Sources/FoundationEssentials/FileManager/FileOperations.swift:354]: I think this will duplicate the call at line 324 above, should we try to avoid asking twice?

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (jmschonfeld) [inline Sources/FoundationEssentials/FileManager/FileOperations.swift:359]: Should we check if we should continue here in case we need to remove siblings?

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (jmschonfeld) [inline Sources/FoundationEssentials/FileManager/FileOperations.swift:349]: Should this return here if one of the above were successful?

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (jmschonfeld) [inline Sources/FoundationEssentials/FileManager/FileOperations.swift:377]: Should we check with the delegate here as well to determine if we should continue?

**Review by jmschonfeld — COMMENTED** [review]
Some minor comments but otherwise this looks great, thanks!

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (compnerd) [reply Sources/FoundationEssentials/FileManager/FileOperations.swift:552]: Yes, we should. I forgot about that.

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (compnerd) [reply Sources/FoundationEssentials/FileManager/FileOperations.swift:359]: Probably

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (compnerd) [reply Sources/FoundationEssentials/FileManager/FileOperations.swift:354]: Yes, the question will be asked twice for the first entry in the case that we hit a non-symlinked directory. We can sink the first check to before the removal attempt.

File: Sources/FoundationEssentials/FileManager/FileOperations.swift
Comment (compnerd) [reply Sources/FoundationEssentials/FileManager/FileOperations.swift:349]: Ah right, thanks!

**Review by jmschonfeld — APPROVED** [review]
LGTM! Running CI just to make sure the conditionals didn't accidentally break the other builds

**Comment by jmschonfeld** [comment]
@swift-ci please test

**Comment by jmschonfeld** [comment]
```
/build/swift-foundation/Sources/FoundationEssentials/FileManager/FileOperations.swift:845:159: error: cannot convert value of type 'UnsafeMutablePointer<CChar>' (aka 'UnsafeMutablePointer<Int8>') to expected argument type 'String'
```

These build failures do look related - it looks like this change breaks the macOS/linux build. Let me know if you need any assistance addressing that before we merge this

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Going to merge this for now - if there are any more comments, happy to address them in a follow up. This should allow us to get to the file diffing changes.

**Pull request merged**
